### PR TITLE
Fixes an infinite running spinner

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
The ApplicationRef propagates never a stable state, when in cinema mode. This is, because in the `cinema.component.ts` asynchronous requests are made, before the app was getting stable.
Also, interval, timeout and similar actions have the same result.

### CODE REVIEW

File: client/src/app/app.component.ts [hunk 1]
```diff
@@ -107,8 +107,7 @@ export class AppComponent {
         appRef.isStable
             .pipe(
                 // take only the stable state
-                filter(s => s),
-                take(1)
+                first(stable => stable)
```
Comment (FinnStutzenstein) [inline client/src/app/app.component.ts:110]: Why did you remove the `take(1)`. This event should only be emitted once.

File: client/src/app/core/core-services/openslides.service.ts [hunk 2]
```diff
@@ -35,6 +35,10 @@ export class OpenSlidesService {
         return this.booted.value;
     }
 
+    public get isStable(): Promise<void> {
```
Comment (FinnStutzenstein) [inline client/src/app/core/core-services/openslides.service.ts:38]: `isStable` suggest, that it is a boolean. Either:
- make `private stable = new Deferred();` to `public readonly stable = new Deferred();`
- rename it to `waitForStability`

Sidenote: Both suggestions (as well as your) does not prevent a user from doing `this.openSlidesService.isStable.resolve()`, which should be prevented (Like a "read-only" Promise). Since I do not know if this easily doable, it can be skipped ;)

File: client/src/app/site/cinema/components/cinema/cinema.component.ts [hunk 3]
```diff
@@ -98,6 +100,15 @@ export class CinemaComponent extends BaseViewComponentDirective implements OnIni
...
         super.setTitle('Autopilot');
+        this.loadSubscriptions();
+    }
+
+    public async toggleListOfSpeakersOpen(): Promise<void> {
+        await this.listOfSpeakersRepo.setListOpenness(this.listOfSpeakers, this.isLosClosed).catch(this.raiseError);
+    }
+
+    private async loadSubscriptions(): Promise<void> {
+        await this.openslidesService.isStable;
```
Comment (FinnStutzenstein) [inline client/src/app/site/cinema/components/cinema/cinema.component.ts:111]: I'm curious why this await helps here. On the first glance no request is send here, isn't it?

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]

File: client/src/app/app.component.ts (hunk 1 above)
Comment (GabrielInTheWorld) [reply client/src/app/app.component.ts]: As of the description from rxjs:

> first
> Emits only the first value. Or emits only the first value that passes some test.

I think, `first` will unify filter and take. ;)

File: client/src/app/core/core-services/openslides.service.ts (hunk 2 above)
Comment (GabrielInTheWorld) [reply client/src/app/core/core-services/openslides.service.ts:38]: Yeah, you're right. I will clean up this, later. :)

File: client/src/app/site/cinema/components/cinema/cinema.component.ts (hunk 3 above)
Comment (GabrielInTheWorld) [reply client/src/app/site/cinema/components/cinema/cinema.component.ts]: You're right. Timeouts, intervals and similar functions have the same result.

**Review by emanuelschuetze — APPROVED** [review]

File: client/src/app/core/core-services/openslides.service.ts (hunk 2 above)
Comment (GabrielInTheWorld) [reply client/src/app/core/core-services/openslides.service.ts:38]: > Sidenote: Both suggestions (as well as your) does not prevent a user from doing `this.openSlidesService.isStable.resolve()`, which should be prevented (Like a "read-only" Promise). Since I do not know if this easily doable, it can be skipped ;)

This is possible, if a user knows, it is a `Deferred`. But a plain `Promise` can't be resolved out of its scope. :)

**Comment by emanuelschuetze** [comment]
@GabrielInTheWorld Thanks.

File: client/src/app/core/core-services/http-interceptors/noop-interceptor.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// import { TestBed } from '@angular/core/testing';
```
Comment (FinnStutzenstein) [inline client/src/app/core/core-services/http-interceptors/noop-interceptor.service.spec.ts:1]: Please just delete this file. Same for the app component test

File: client/src/app/core/core-services/openslides.service.ts (hunk 2 above)
Comment (FinnStutzenstein) [reply client/src/app/core/core-services/openslides.service.ts:38]: Ah, right

Comment (FinnStutzenstein) [reply client/src/app/core/core-services/openslides.service.ts:38]: Still a todo

File: client/src/app/site/projector/components/projector-detail/projector-detail.component.ts [hunk 5]
```diff
@@ -353,4 +350,16 @@ export class ProjectorDetailComponent extends BaseViewComponentDirective impleme
             this.messageRepo.create(sendData).then(() => {}, this.raiseError);
         }
     }
+
+    private async loadSubscriptions(): Promise<void> {
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/projector-detail/projector-detail.component.ts:354]: Do not wait until an stable app to subscribe. These subscriptions itself do not keep the spinner up, but some other effects. This is too broad...

File: client/src/app/site/projector/components/projector-detail/projector-detail.component.ts [hunk 6]
```diff
@@ -353,4 +350,16 @@ export class ProjectorDetailComponent extends BaseViewComponentDirective impleme
...
+        await this.stableService.isStable;
+        this.countdownRepo.getViewModelListObservable().subscribe(countdowns => (this.countdowns = countdowns));
+        this.messageRepo.getViewModelListObservable().subscribe(messages => (this.messages = messages));
+        this.repo.getViewModelListObservable().subscribe(projectors => (this.projectorCount = projectors.length));
+    }
+
+    private async installUpdater(): Promise<void> {
+        await this.stableService.isStable;
+        this.subscriptions.push(timer(0, 500).subscribe(() => this.cd.detectChanges()));
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/projector-detail/projector-detail.component.ts:363]: This seems hacky. Why do you need a detect changes 500ms after the app stable event? Please put the `detectChanges` to the place, where it is actually needed.

As a sidenote: You can use https://github.com/OpenSlides/OpenSlides/blob/master/client/src/app/core/promises/sleep.ts: `await SleepPromise(500)` if you want to do that kind of waiting.

File: client/src/app/site/projector/components/projector-list/projector-list.component.ts [hunk 7]
```diff
@@ -138,4 +133,16 @@ export class ProjectorListComponent extends BaseViewComponentDirective implement
         super.ngOnDestroy();
         this.cd.detach();
     }
+
+    private async installUpdater(): Promise<void> {
+        await this.openslidesService.isStable;
+        /**
+         * Angulars change detection goes nuts, since countdown and motios with long texts are pushing too much data
+         */
+        this.subscriptions.push(
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/projector-list/projector-list.component.ts:142]: Please use the SleepPromise.

File: client/src/app/core/core-services/stable.service.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+
+import { BehaviorSubject, Observable } from 'rxjs';
+
+import { Deferred } from '../promises/deferred';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class StableService {
```
Comment (FinnStutzenstein) [inline client/src/app/core/core-services/stable.service.ts:10]: Integrate this into the OpenSlidesStatusService. Use this service only, so remove the `isStable` in the OpenSlides Service

File: client/src/app/core/core-services/stable.service.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    public get isStable(): Promise<void> {
+        return this.stable;
+    }
+
+    public get booted(): Observable<boolean> {
```
Comment (FinnStutzenstein) [inline client/src/app/core/core-services/stable.service.ts:15]: This is only needed in the interceptor - move the BehaviorSubject there

File: client/src/app/core/core-services/http-interceptors/noop-interceptor.service.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+import { Observable, Subject } from 'rxjs';
+import { first, mergeMap } from 'rxjs/operators';
+
+import { StableService } from '../stable.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class NoopInterceptorService implements HttpInterceptor {
```
Comment (FinnStutzenstein) [inline client/src/app/core/core-services/http-interceptors/noop-interceptor.service.ts:12]: Is it a Noop interceptor?

File: client/src/app/core/core-services/stable.service.ts (hunk 9 above)
Comment (FinnStutzenstein) [reply client/src/app/core/core-services/stable.service.ts:15]: Also: booted is another terminology that indicates, that OpenSlides has started up - do not use this wording twice.

**Review by FinnStutzenstein — COMMENTED** [review]
@emanuelschuetze Please wait for reviews!!!

**Comment by FinnStutzenstein** [comment]
@GabrielInTheWorld This needs some fixes.

File: client/src/app/app.component.spec.ts [hunk 11]
```diff
@@ -1,35 +1,33 @@
-import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
+// import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
```
Comment (FinnStutzenstein) [inline client/src/app/app.component.spec.ts:1]: Please just remove this file.
